### PR TITLE
Dc/da 3099

### PR DESCRIPTION
**Pull request by xtoller**
## Resolves *[ticket #](link to ticket)*
DA-3099 https://precisionmedicineinitiative.atlassian.net/browse/DA-3099

## Description of changes/additions
NPH Participant API Implementation

## Tests
- [] unit tests

### CODE REVIEW

File: rdr_service/api/nph_participant_api.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    Return nph_participant information for all records or a subset based
+    on the awardee parameter.
+    """
+
+    try:
+        query = request.get_data().decode('utf8')
+        validate_error = validate_query(query)
+        if validate_error:
+            log_api_request(log=request.log_record)
```
Comment (kenny-skaggs) [inline rdr_service/api/nph_participant_api.py]: I might be missing something, but why call `log_api_request` again here when it's already called in `auth_required` (the decorator on the method).

File: rdr_service/api/nph_participant_api_schemas/schema.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+
+class EventCollection(ObjectType):
+    current = SortableField(Event)
+    # TODO: historical field need to sort by newest to oldest for a given aspect of a participant’s data
+    historical = List(Event)
+
+    @staticmethod
+    def sort(context, value):
+        pass
```
Comment (kenny-skaggs) [inline rdr_service/api/nph_participant_api_schemas/schema.py]: This should raise an error for now if a client tries to sort by `historical`

File: rdr_service/api/nph_participant_api_schemas/schema.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+
+class Sample(ObjectType):
+    parent = SortableField(EventCollection)
+    child = SortableField(EventCollection)  # todo: join 'child' on parent ('sample') and sort_table = 'child'
```
Comment (kenny-skaggs) [inline rdr_service/api/nph_participant_api_schemas/schema.py]: This todo seems like it's already implemented below, was it left in by accident or is it still to remind of something?

File: requirements_base.txt [hunk 4]
```diff
@@ -121,3 +121,8 @@ wrapt==1.12.1             # via astroid
 xmltodict==0.12.0         # via -r requirements.in
 zope.event==4.4           # via gevent
 zope.interface==5.1.0     # via gevent
+# DA-3099 -- dchan
+Flask-GraphQL==2.0.0
+graphene==3.1.0
+graphql-core==3.1
+names
```
Comment (kenny-skaggs) [inline requirements_base.txt]: Are any of these dependencies needed other than `graphene`? I think `graphene` brings in `graphql-core` on it's own, I'm not seeing where `flask-graphql` was used, and I think `names` was something from earlier testing, right?

File: rdr_service/api/nph_participant_api.py (hunk 1 above)
Comment (xtoller) [reply rdr_service/api/nph_participant_api.py]: change to use logging.error

File: rdr_service/api/nph_participant_api_schemas/schema.py (hunk 2 above)
Comment (xtoller) [reply rdr_service/api/nph_participant_api_schemas/schema.py]: raise an error when client request to sort using hisotrical

File: rdr_service/api/nph_participant_api_schemas/schema.py (hunk 3 above)
Comment (xtoller) [reply rdr_service/api/nph_participant_api_schemas/schema.py]: removed the TODO item

File: requirements_base.txt (hunk 4 above)
Comment (xtoller) [reply requirements_base.txt]: removed unused package in requirements_base.txt

**Review by kenny-skaggs — APPROVED** [review]

**Review by brianmkimmel — APPROVED** [review]

**Comment by xtoller** [comment]
Merging dc/DA-3099 to devel (NPH Participant API)

**Pull request merged**